packages/sales-api-service/src/server/routes/__tests__/renewals.spec.js

```diff
@@ -0,0 +1,111 @@
+  beforeEach(jest.clearAllMocks)
+
+  it('should call permissionForFullReferenceNumber with referenceNumber', async () => {
+    executeQuery.mockResolvedValueOnce([permissionForFullReferenceNumberMock()])
+
+    const referenceNumber = 'REFERENCE123'
+    const request = getMockRequest(referenceNumber)
+    const responseToolkit = getMockResponseToolkit()
+
+    await permissionRenewalData[0].options.handler(request, responseToolkit)
+
+    expect(permissionForFullReferenceNumber).toHaveBeenCalledWith(referenceNumber)
+  })
+
+  it('should call preparePermissionDataForRenewal with the expected data', async () => {
+    const permissionMock = permissionForFullReferenceNumberMock()
+    executeQuery.mockResolvedValueOnce([permissionMock])
+
+    await permissionRenewalData[0].options.handler(getMockRequest(), getMockResponseToolkit())
+
+    const expectedData = {
+      ...permissionMock.entity.toJSON(),
+      licensee: permissionMock.expanded.licensee.entity.toJSON(),
+      concessions: [],
+      permit: permissionMock.expanded.permit.entity.toJSON()
+    }
+    expect(preparePermissionDataForRenewal).toHaveBeenCalledWith(expectedData)
+  })
+
+  it('should return continue response', async () => {
+    executeQuery.mockResolvedValueOnce([permissionForFullReferenceNumberMock()])
+    const request = getMockRequest({})
+    const responseToolkit = getMockResponseToolkit()
+
+    expect(await permissionRenewalData[0].options.handler(request, responseToolkit)).toEqual(responseToolkit.continue)
+  })
+
+  describe('if there are no matching permissions', () => {
+    it('should raise an error', async () => {
+      executeQuery.mockResolvedValueOnce([])
+      const request = getMockRequest({})
+      const responseToolkit = getMockResponseToolkit()
+
+      const boomError = Boom.unauthorized('Permission not found for renewal')
+      await expect(permissionRenewalData[0].options.handler(request, responseToolkit)).rejects.toThrow(boomError)
+    })
+  })
+
+  describe('if there are multiple matching permissions', () => {
+    it('should raise an error', async () => {
+      executeQuery.mockResolvedValueOnce(['foo', 'bar'])
+      const request = getMockRequest({})
+      const responseToolkit = getMockResponseToolkit()
+
+      const error = 'Unable to get permission data for renewal, non-unique results for query'
+      await expect(permissionRenewalData[0].options.handler(request, responseToolkit)).rejects.toThrow(error)
+    })
+  })
+
+  it('should throw error if query execution fails', async () => {
+    permissionForFullReferenceNumber.mockResolvedValueOnce({ filter: 'coffee' })
+    const error = 'Wuh-oh!'
+    executeQuery.mockImplementationOnce(() => {
+      throw new Error(error)
+    })
+    const request = getMockRequest({})
+    const responseToolkit = getMockResponseToolkit()
+
+    await expect(permissionRenewalData[0].options.handler(request, responseToolkit)).rejects.toThrow(error)
+  })
```

Comment:
to cover the other parts of the export from renewals, I think you could use a snapshot test covering description, notes, tags, etc